src/commonMain/kotlin/fr/acinq/lightning/payment/OfferManager.kt

```diff
@@ -0,0 +1,142 @@
+                    val remoteChannelUpdates = channels.values.mapNotNull { channelState ->
+                        when (channelState) {
+                            is Normal -> channelState.remoteChannelUpdate
+                            is Offline -> (channelState.state as? Normal)?.remoteChannelUpdate
+                            is Syncing -> (channelState.state as? Normal)?.remoteChannelUpdate
+                            else -> null
+                        }
+                    }
+                    val paymentInfo = OfferTypes.PaymentInfo(
+                        feeBase = remoteChannelUpdates.maxOfOrNull { it.feeBaseMsat } ?: walletParams.invoiceDefaultRoutingFees.feeBase,
+                        feeProportionalMillionths = remoteChannelUpdates.maxOfOrNull { it.feeProportionalMillionths } ?: walletParams.invoiceDefaultRoutingFees.feeProportional,
+                        cltvExpiryDelta = remoteChannelUpdates.maxOfOrNull { it.cltvExpiryDelta } ?: walletParams.invoiceDefaultRoutingFees.cltvExpiryDelta,
+                        minHtlc = remoteChannelUpdates.minOfOrNull { it.htlcMinimumMsat } ?: 1.msat,
+                        maxHtlc = amount,
+                        allowedFeatures = Features.empty
+                    )
+                    val remoteNodePayload = RouteBlindingEncryptedData(TlvStream(
+                        RouteBlindingEncryptedDataTlv.OutgoingChannelId(ShortChannelId.peerId(nodeParams.nodeId)),
+                        RouteBlindingEncryptedDataTlv.PaymentRelay(paymentInfo.cltvExpiryDelta, paymentInfo.feeProportionalMillionths, paymentInfo.feeBase),
+                        RouteBlindingEncryptedDataTlv.PaymentConstraints((paymentInfo.cltvExpiryDelta + nodeParams.maxFinalCltvExpiryDelta).toCltvExpiry(currentBlockHeight.toLong()), paymentInfo.minHtlc)
+                    )).write().toByteVector()
+                    val blindedRoute = RouteBlinding.create(randomKey(), listOf(remoteNodeId, nodeParams.nodeId), listOf(remoteNodePayload, recipientPayload)).route
+                    val path = Bolt12Invoice.Companion.PaymentBlindedContactInfo(OfferTypes.ContactInfo.BlindedPath(blindedRoute), paymentInfo)
+                    val invoice = Bolt12Invoice(request, preimage, decrypted.blindedPrivateKey, nodeParams.bolt12invoiceExpiry.inWholeSeconds, nodeParams.features.bolt12Features(), listOf(path))
+                    return when (val invoiceMessage = buildMessage(randomKey(), randomKey(), listOf(IntermediateNode(remoteNodeId)), Destination.BlindedPath(decrypted.content.replyPath), TlvStream(OnionMessagePayloadTlv.Invoice(invoice.records)))) {
+                        is Left -> null
+                        is Right -> OnionMessageAction.SendMessage(invoiceMessage.value)
+                    }
+                } else {
+                    // TODO: send back invoice error
```

Comment:
We should log something whenever we return `null` in this function to help debug.